ENGAGEHF/HeartHealth/Shared/GraphSection/VitalsGraph.swift

```diff
@@ -29,15 +29,8 @@ struct VitalsGraph: View {
             // Overlay for tracking gestures
             .modifier(GestureOverlay(viewModel: viewModel))
             // State change modifiers to listen for updates to the environment and handle errors
-            .onChange(of: data) {
-                viewModel.processData(data, options: options)
-            }
-            .onChange(of: options) {
-                viewModel.processData(data, options: options)
-            }
-            .onAppear {
-                viewModel.processData(data, options: options)
-            }
+            .onChange(of: options) { viewModel.processData(data, options: options) }
```

Comment:
@PSchmiedmayer For this .onChange modifier, options (which has type VitalsGraphOptions) must conform to Equatable. The spirit of this modifier is to update the data whenever new options are passed to the view, so I conform VitalsGraphOptions to Equatable by simply seeing if the ID has changed:
```swift
static func == (lhs: VitalsGraphOptions, rhs: VitalsGraphOptions) -> Bool {
     lhs.id == rhs.id
}
``` 
This may cause us to process the data more often than we want, since this means that the onChange closure will be called whenever a new options struct is passed to the view (not necessarily when the options themselves are different). I thought I'd point this (and in general how we decide when to re-process the data) out as an area we might refine in the future.